src/server/DeviceServer.js

```diff
@@ -327,6 +319,37 @@ class DeviceServer {
     await this._eventPublisher.publish(eventData);
   };
 
+  _onDeviceClaimCodeMessage = async (
+    message: Message,
+    device: Device,
+  ): Promise<void> => {
+    const claimCode = message.getPayload().toString();
+    const deviceID = device.getID();
+    const deviceAttributes = await this._deviceAttributeRepository.getById(deviceID);
+    if (!deviceAttributes) {
+      return;
+    }
+    if (deviceAttributes.ownerID) {
+      return;
+    }
+
+    // todo if we figure out how to delete claimCode from the device
+    // we could make the flow more clean.
+    if (deviceAttributes.claimCode !== claimCode) {
+      const claimRequestUser = await this._userRepository.getByClaimCode(claimCode);
+      if (!claimRequestUser) {
+        return;
+      }
+
+      await this._userRepository.removeClaimCode(claimRequestUser.id, claimCode);
```

Comment:
Yeah, that is a good point.  

I think we also need an expiration time on the claim-codes and they get cleaned up after a certain amount of time.

I don't really like the idea of a user spamming the claims API and ending up with hundreds or thousands of claim IDs in their user object... It might make sense just to store the claim codes in memory so this doesn't happen.